src/main/java/org/testng/reporters/EmailableReporter.java

```diff
@@ -73,8 +85,12 @@ public void generateReport(List<XmlSuite> xml, List<ISuite> suites, String outdi
 
   protected PrintWriter createWriter(String outdir) throws IOException {
     new File(outdir).mkdirs();
-    return new PrintWriter(new BufferedWriter(new FileWriter(new File(outdir,
-        "emailable-report.html"))));
+    String jvmArg = System.getProperty(JVM_ARG);
+    if ((jvmArg != null) && (!jvmArg.trim().isEmpty())) {
+      fileName = jvmArg;
+    }
+
```

Comment:
the `fileName` parameter is overwritten only if its not null or empty. In other cases we use the default value (or) the value passed via the `ReporterConfig`